src/Compilers/CSharp/Portable/Lowering/DiagnosticsPass_ExpressionTrees.cs

```diff
@@ -211,6 +211,18 @@ public override BoundNode VisitCompoundAssignmentOperator(BoundCompoundAssignmen
             }
         }
 
+        public override BoundNode Visit(BoundNode node)
+        {
+            if (_inExpressionLambda &&
```

Comment:
I'd like to be able to just do this instead of having to add Visit overrides and helper calls for every type of expression syntax that could have a ByRefLike type.

Reply:
I'd suggest something like

```C#
override BoundNode Visit(BoundExpression expr)
{
   if (expr.Type?.IsByRefLikeType == true)
   {
        Error(...)
   }
   return base.Visit(node);
}
```